src/ivectorbin/compute-vad-from-frame-likes.cc

```diff
@@ -0,0 +1,186 @@
+                unordered_map<int32, int32> *map) {
+  Input map_input(map_rxfilename);
+  for (int32 i = 0; i < num_classes; i++)
+    (*map)[i] = i;
+
+  if (!map_rxfilename.empty()) {
+    std::string line;
+    while (std::getline(map_input.Stream(), line)) {
+      std::vector<std::string> fields;
+      SplitStringToVector(line, " \t\n\r", true, &fields);
+      if (fields.size() != 2) {
+        KALDI_ERR << "Bad line. Expected two fields, got: "
+                  << line;
+      }
+      (*map)[std::atoi(fields[0].c_str())] = std::atoi(fields[1].c_str());
+    }
+  }
+
+  if (map->size() > num_classes)
+    KALDI_ERR << "Map table has " << map->size() << " classes.  "
+              << "Expected " << num_classes << " or fewer";
+}
+
+void PreparePriors(const std::string &priors_rxfilename, int32 num_classes,
+                unordered_map<int32, BaseFloat> *priors) {
+  Input priors_input(priors_rxfilename);
+  if (priors_rxfilename.empty()) {
+    for (int32 i = 0; i < num_classes; i++)
+      (*priors)[i] = 0;
+  } else {
+    std::string line;
+    while (std::getline(priors_input.Stream(), line)) {
+      std::vector<std::string> fields;
+      SplitStringToVector(line, " \t\n\r", true, &fields);
+      if (fields.size() != 2) {
+        KALDI_ERR << "Bad line. Expected two fields, got: "
+                  << line;
+      }
+        (*priors)[std::atoi(fields[0].c_str())] =
+          std::atof(fields[1].c_str());
+    }
+  }
+  if (priors->size() != num_classes)
+    KALDI_ERR << "Prior table has " << priors->size() << " classes.  "
+              << "Expected " << num_classes;
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    const char *usage =
+      "This program computes frame-level voice activity decisions from a\n"
+      "set of input frame-level log-likelihoods.  Frames are assigned labels\n"
+      "according to the class for which the log-likelihood (optionally\n"
+      "weighted by a prior) is maximal.  The class labels are determined\n"
+      "by the order of inputs on the command line. See options for more\n"
+      "details.\n"
+      "Usage: compute-vad-from-frame-likes [options] <likes-rspecifier-1>\n"
+      "    ... <likes-rspecifier-n> <vad-wspecifier>\n"
+      "e.g.: compute-vad-from-frame-likes --map=label_map.txt\n"
+      "    scp:likes1.scp scp:likes2.scp ark:vad.ark\n";
+
+    ParseOptions po(usage);
+    std::string map_rxfilename;
+    std::string priors_rxfilename;
+
+    po.Register("map", &map_rxfilename, "Table that defines the frame-level"
+    " labels.  For each row, the first field is the zero-based index of the"
+    " input likelihood archive and the second field is the associated"
+    " integer label.");
+
+    po.Register("priors", &priors_rxfilename, "Table that specifies"
+    " log-priors for each class. For each row, the first field is the"
+    " zero-based index of the input likelihood archive and the second"
+    " field is the log-prior.");
```

Comment:
It seems to me that you would always have a file like
0 something
1 something
2 something
and therefore the 0 1 and 2 are redundant.  Also, wouldn't it be easier to just specify something like
  --priors=0.1,0.8,0.1
i.e. a comma-separated vector (colon-separation is now deprecated), which you can parse using SplitStringToFloats to a vector of float?  This will avoid the hassle of a file.